src/components/periodic-table/PeriodicTable.tsx

```diff
@@ -1,177 +1,101 @@
-        <div className="periodic-table__cell periodic-table__cell--label">
-          3
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          4
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          5
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          6
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          7
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          8
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          9
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          10
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          11
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          12
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          13
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          14
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          15
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          16
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          17
-        </div>
-        <div className="periodic-table__cell periodic-table__cell--label">
-          18
-        </div>
-      </div>,
-    ];
-
-    for (let i = 0; i < data.length; i++) {
-      const row = data[i];
-
-      rows.push(
-        <div className="periodic-table__row" key={`row-${i}`}>
-          {i <= 6 ? (
-            <div className="periodic-table__cell periodic-table__cell--label">
-              {i + 1}
-            </div>
-          ) : (
-            <div className="periodic-table__cell periodic-table__cell--label" />
-          )}
-
-          {row.map((element, index) => {
-            if (element <= 0) {
-              return (
-                <div
-                  key={`row-${i}-cell-${index}`}
-                  className="periodic-table__cell periodic-table__cell--empty"
-                />
-              );
-            }
-
-            const ptElement = this.props.elementRenderer(element);
-            const $component = ptElement.component;
-            const props = ptElement.props;
-
-            return (
-              <div
-                className="periodic-table__cell"
-                key={`row-${i}-cell-${index}`}
-              >
-                <$component {...props} />
-              </div>
-            );
-          })}
-        </div>
-      );
-    }
-
-    return rows;
-  }
+  return <div className="periodic-table">{buildTable(elementRenderer)}</div>;
```

Comment:
I wonder if this would be better to split into components rather than extracting a function, something like:

```ts
return (
  <div className="periodic-table">
    <PeriodicTableHeader />
    <PerididicTableRows elementRenderer={elementRenderer} />
  </div>
);

Reply:
Yes I was thinking the same. I have refactored `mass-calculator` splitting in more components. In this case, I thought to leave it as is because the component was simpler. In any case, if splitting into a separate component makes the thing clearer I agree with you

Reply:
Happy to take as-is, just thinking aloud